app/routes/cdo/create/microchip-search.js

```diff
@@ -75,3 +78,15 @@ const duplicateMicrochipsInSession = (request, details) => {
 const generateDuplicateError = message => {
   return new Joi.ValidationError(message, [{ message, path: ['microchipNumber'] }])
 }
+
+const getLatestDogIndex = (request) => {
+  if (getDogs(request)) {
+    return getDogs(request)?.filter(x => x.applicationType).length + 1
+  }
+
+  return 1
+}
+
+const determineDogId = (request, details) => {
```

Comment:
not one for this PR, but would probably be good to refactor this a bit so logic is clearer